### ISSUE TITLE
/data/local/tmp/frida-server[1]: syntax error: unexpected '('

### ISSUE THREAD

**Issue by solvorry** [frida/frida#2683]
Kindly someone who is using Macbook M2 please guide me how to resolve the same.

adb shell getprop ro.product.cpu.abi                                                  
arm64-v8a

This frida-server I have downloaded
https://github.com/frida/frida/releases/download/16.1.5/frida-server-16.1.5-macos-arm64.xz

Getting following error
/data/local/tmp/frida-server[1]: syntax error: unexpected '('

Working fine with
frida-ps -Uai

Unable to run this as well
frida -U -f com.xxx.xxx -l universal-android-ssl-pinning-bypass-with-frida.js

Thanks & Warm Regards.

**Comment by jpstotz** [comment]
@solvorry What do you want to do? `frida -U` is only required if you want to connect to an iPhone connected via USB (which would then require to run frida-server for iOS on the target device).

**Comment by solvorry** [comment]
No I want it for Genymotion Android Device

**Comment by jpstotz** [comment]
Yes Android devices and emulator also need `-U` flag. But then you need to install `frida-server` inside the emulator and therefore you need the package [frida-server-16.1.7-android-x86_64.xz](https://github.com/frida/frida/releases/download/16.1.7/frida-server-16.1.7-android-x86_64.xz) (or may be the 32bit "x86" variant without "_64" depending on the used Genymotion image.

**Comment by cdfhalle** [comment]
I have a similar Issue, I have android studio emulator running (Pixel 3, API 28, Android 9.0 Pie, arm64) and tried to install frida-server-16.2.1-macos-arm64. I followed all the steps given on the [Frida Documentation Page](https://frida.re/docs/android/) but when I execute `adb shell "/data/local/tmp/frida-server &"` I receive the exact same error as mentioned above. In case that is of any importance, I'm running the emulator in the MacOS operating system with an M3 chip.

**Comment by arunstar** [comment]
For Genymotion you need to use: [frida-server-16.4.8-android-arm64.xz](https://github.com/frida/frida/releases)
